cli/src/program_v4.rs

```diff
@@ -1140,10 +1169,13 @@ fn send_messages(
     }
 
     for message in final_messages.into_iter() {
-        send_message(
+        let result = send_message(
             message,
             &[config.signers[0], config.signers[*auth_signer_index]],
         )?;
+        if additional_cli_config.sign_only {
```

Comment:
Could there be multiple messages that need signing? If so, it'll return here after the message, right?

Reply:
Yes, I think this is a general problem in the CLI offline signing mechanism: It can only return one signature per invocation. Thus, we need a two step deployment so that only the final replacement of the program with a buffer is signed offline and all the TXs for the upload before are done using online signing.